src/basline_rule_syntax_checker.py

```diff
@@ -0,0 +1,125 @@
+import re
+from selector import IpSelector
+
+
+class RuleSyntaxChecker:
+    """
+    This class make syntax and validity checks on baseline rule's fields
+    """
+
+    def __init__(self):
+        self.rule_name = ''
```

Comment:
I think it will be nicer if all methods in this class will be static, and the class will have no data members.
The `rule_name` context can be provided by a try-except in `BaselineRule.__init__()`